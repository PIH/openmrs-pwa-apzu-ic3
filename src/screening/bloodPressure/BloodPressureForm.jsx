import React from "react";
import {Submit, Obs, Section, EncounterFormPage, formValidations } from '@openmrs/react-components';
import {Grid, Row, FormGroup, ControlLabel, Col} from 'react-bootstrap';
import { ENCOUNTER_TYPES, CONCEPTS } from "../../constants";
import {colHeight, leftTextAlign} from "../../pwaStyles";

/**
 * Range of possible values
 * SBP 50-260
 * DBP 40-140
 *
 * Abnormal results
 * SBP > 160
 * DBP > 110
 */
const minValue40 = formValidations.minValue(40);
const minValue50 = formValidations.minValue(50);
const maxValue140 = formValidations.maxValue(140);
const maxValue260 = formValidations.maxValue(260);

const abnormalMaxValue110 = formValidations.abnormalMaxValue(110);
const abnormalMaxValue160 = formValidations.abnormalMaxValue(160);

let BloodPressureForm = props => {

  const formContent = (
    <Grid>
      <Section title="Enter Patient Blood Pressure"></Section>
      <Row>
        <Col sm={20} md={20} style={ colHeight }>
          <span><h1>{ '' }</h1></span>
        </Col>
      </Row>
      <Row>
        <FormGroup controlId="formSystolic">
          <Col componentClass={ControlLabel} sm={4}>
            Systolic Blood Pressure
          </Col>
          <Col sm={2}>
            <Obs
              concept={CONCEPTS.SystolicBloodPressure.uuid}
              placeholder="Systolic value"
              path="systolic"
              validate={[minValue50, maxValue260]}
              warn={ abnormalMaxValue160 }
            />
          </Col>
          <Col componentClass={ControlLabel} sm={2} style={ leftTextAlign }>
            mmHG
          </Col>
        </FormGroup>
      </Row>
      <Row>
        <FormGroup controlId="formDiastolic">
          <Col componentClass={ControlLabel} sm={4}>
            Diastolic Blood Pressure
          </Col>
          <Col sm={2}>
            <Obs
              concept={CONCEPTS.DiastolicBloodPressure.uuid}
              placeholder="Diastolic value"
              validate={[minValue40, maxValue140]}
              path="diastolic"
              warn={ abnormalMaxValue110 }
            />
          </Col>
          <Col componentClass={ControlLabel} sm={2} style={ leftTextAlign }>
            mmHG
          </Col>
        </FormGroup>
      </Row>
      <Row>
        <Col sm={4} xsOffset={5}>
          <Submit/>
        </Col>
      </Row>
    </Grid>
  );

  return (
    <EncounterFormPage
      afterSubmitLink="/screening/bloodPressure/queue"
      backLink="/screening/bloodPressure/queue"
      encounterType={ENCOUNTER_TYPES.BloodPressureEncounterType}
      formContent={formContent}
      title="Blood Pressure"
      formId="bp-form"
    />
  );
};

export default BloodPressureForm;
