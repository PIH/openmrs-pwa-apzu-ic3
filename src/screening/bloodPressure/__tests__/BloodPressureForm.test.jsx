import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import MockRouter from 'react-mock-router';
import { Provider } from 'react-redux';
import BloodPressureForm from '../BloodPressureForm';


let props, store;
let mountedComponent;

const mockStore = configureMockStore();

const bloodPressureForm = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <Provider store={store}>
        <MockRouter>
          <BloodPressureForm {...props} />
        </MockRouter>
      </Provider>);
  }
  return mountedComponent;
};

describe('Component: BloodPressureForm', () => {
  beforeEach(() => {
    props = {};
    store = mockStore(
      {
        openmrs: {
          patients: {
            "abc": {
              uuid: "abc",
              visit: {
                uuid: "def"
              }
            }
          },
          selectedPatient: "abc"
        }
      });
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    expect(toJson(bloodPressureForm())).toMatchSnapshot();
  });

});
