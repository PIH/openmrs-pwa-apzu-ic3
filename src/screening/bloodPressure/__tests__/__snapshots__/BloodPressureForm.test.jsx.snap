// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BloodPressureForm renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MockRouter
    createHref={[Function]}
    location=""
    params={Object {}}
    path=""
    push={[Function]}
    replace={[Function]}
    url=""
  >
    <BloodPressureForm>
      <Connect(ScreeningForm)
        encounterType={
          Object {
            "uuid": "0C36F6FB-660E-485F-AF04-249579C9EAC9",
          }
        }
        formContent={
          <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
          >
            <Section
              title="Enter Patient Blood Pressure"
            />
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                md={20}
                sm={20}
                style={
                  Object {
                    "height": "40px",
                  }
                }
              >
                <span>
                  <h1 />
                </span>
              </Col>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
                controlId="formSystolic"
              >
                <Col
                  bsClass="col"
                  componentClass={[Function]}
                  sm={4}
                >
                  Systolic Blood Pressure
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  sm={2}
                >
                  <WithFormContext(Connect(Obs))
                    concept="6569bffe-977f-11e1-8993-905e29aff6c1"
                    path="systolic"
                    placeholder="Systolic value"
                    validate={
                      Array [
                        [Function],
                        [Function],
                      ]
                    }
                    warn={[Function]}
                  />
                </Col>
                <Col
                  bsClass="col"
                  componentClass={[Function]}
                  sm={2}
                  style={
                    Object {
                      "textAlign": "left",
                    }
                  }
                >
                  mmHG
                </Col>
              </FormGroup>
            </Row>
            <Row
              bsClass="row"
              componentClass="div"
            >
              <FormGroup
                bsClass="form-group"
                controlId="formDiastolic"
              >
                <Col
                  bsClass="col"
                  componentClass={[Function]}
                  sm={4}
                >
                  Diastolic Blood Pressure
                </Col>
                <Col
                  bsClass="col"
                  componentClass="div"
                  sm={2}
                >
                  <WithFormContext(Connect(Obs))
                    concept="6569c116-977f-11e1-8993-905e29aff6c1"
                    path="diastolic"
                    placeholder="Diastolic value"
                    validate={
                      Array [
                        [Function],
                        [Function],
                      ]
                    }
                    warn={[Function]}
                  />
                </Col>
                <Col
                  bsClass="col"
                  componentClass={[Function]}
                  sm={2}
                  style={
                    Object {
                      "textAlign": "left",
                    }
                  }
                >
                  mmHG
                </Col>
              </FormGroup>
            </Row>
          </Grid>
        }
        formId="bp-form"
        title="Blood Pressure"
      >
        <ScreeningForm
          dispatch={[Function]}
          encounterType={
            Object {
              "uuid": "0C36F6FB-660E-485F-AF04-249579C9EAC9",
            }
          }
          formContent={
            <Grid
              bsClass="container"
              componentClass="div"
              fluid={false}
            >
              <Section
                title="Enter Patient Blood Pressure"
              />
              <Row
                bsClass="row"
                componentClass="div"
              >
                <Col
                  bsClass="col"
                  componentClass="div"
                  md={20}
                  sm={20}
                  style={
                    Object {
                      "height": "40px",
                    }
                  }
                >
                  <span>
                    <h1 />
                  </span>
                </Col>
              </Row>
              <Row
                bsClass="row"
                componentClass="div"
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="formSystolic"
                >
                  <Col
                    bsClass="col"
                    componentClass={[Function]}
                    sm={4}
                  >
                    Systolic Blood Pressure
                  </Col>
                  <Col
                    bsClass="col"
                    componentClass="div"
                    sm={2}
                  >
                    <WithFormContext(Connect(Obs))
                      concept="6569bffe-977f-11e1-8993-905e29aff6c1"
                      path="systolic"
                      placeholder="Systolic value"
                      validate={
                        Array [
                          [Function],
                          [Function],
                        ]
                      }
                      warn={[Function]}
                    />
                  </Col>
                  <Col
                    bsClass="col"
                    componentClass={[Function]}
                    sm={2}
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    mmHG
                  </Col>
                </FormGroup>
              </Row>
              <Row
                bsClass="row"
                componentClass="div"
              >
                <FormGroup
                  bsClass="form-group"
                  controlId="formDiastolic"
                >
                  <Col
                    bsClass="col"
                    componentClass={[Function]}
                    sm={4}
                  >
                    Diastolic Blood Pressure
                  </Col>
                  <Col
                    bsClass="col"
                    componentClass="div"
                    sm={2}
                  >
                    <WithFormContext(Connect(Obs))
                      concept="6569c116-977f-11e1-8993-905e29aff6c1"
                      path="diastolic"
                      placeholder="Diastolic value"
                      validate={
                        Array [
                          [Function],
                          [Function],
                        ]
                      }
                      warn={[Function]}
                    />
                  </Col>
                  <Col
                    bsClass="col"
                    componentClass={[Function]}
                    sm={2}
                    style={
                      Object {
                        "textAlign": "left",
                      }
                    }
                  >
                    mmHG
                  </Col>
                </FormGroup>
              </Row>
            </Grid>
          }
          formId="bp-form"
          patient={
            Object {
              "uuid": "abc",
              "visit": Object {
                "uuid": "def",
              },
            }
          }
          title="Blood Pressure"
        >
          <Connect(EncounterFormPage)
            encounterRole={
              Object {
                "uuid": "a0b03050-c99b-11e0-9572-0800200c9a66",
              }
            }
            encounterType={
              Object {
                "uuid": "0C36F6FB-660E-485F-AF04-249579C9EAC9",
              }
            }
            formContent={
              <Grid
                bsClass="container"
                componentClass="div"
                fluid={false}
              >
                <Section
                  title="Enter Patient Blood Pressure"
                />
                <Row
                  bsClass="row"
                  componentClass="div"
                >
                  <Col
                    bsClass="col"
                    componentClass="div"
                    md={20}
                    sm={20}
                    style={
                      Object {
                        "height": "40px",
                      }
                    }
                  >
                    <span>
                      <h1 />
                    </span>
                  </Col>
                </Row>
                <Row
                  bsClass="row"
                  componentClass="div"
                >
                  <FormGroup
                    bsClass="form-group"
                    controlId="formSystolic"
                  >
                    <Col
                      bsClass="col"
                      componentClass={[Function]}
                      sm={4}
                    >
                      Systolic Blood Pressure
                    </Col>
                    <Col
                      bsClass="col"
                      componentClass="div"
                      sm={2}
                    >
                      <WithFormContext(Connect(Obs))
                        concept="6569bffe-977f-11e1-8993-905e29aff6c1"
                        path="systolic"
                        placeholder="Systolic value"
                        validate={
                          Array [
                            [Function],
                            [Function],
                          ]
                        }
                        warn={[Function]}
                      />
                    </Col>
                    <Col
                      bsClass="col"
                      componentClass={[Function]}
                      sm={2}
                      style={
                        Object {
                          "textAlign": "left",
                        }
                      }
                    >
                      mmHG
                    </Col>
                  </FormGroup>
                </Row>
                <Row
                  bsClass="row"
                  componentClass="div"
                >
                  <FormGroup
                    bsClass="form-group"
                    controlId="formDiastolic"
                  >
                    <Col
                      bsClass="col"
                      componentClass={[Function]}
                      sm={4}
                    >
                      Diastolic Blood Pressure
                    </Col>
                    <Col
                      bsClass="col"
                      componentClass="div"
                      sm={2}
                    >
                      <WithFormContext(Connect(Obs))
                        concept="6569c116-977f-11e1-8993-905e29aff6c1"
                        path="diastolic"
                        placeholder="Diastolic value"
                        validate={
                          Array [
                            [Function],
                            [Function],
                          ]
                        }
                        warn={[Function]}
                      />
                    </Col>
                    <Col
                      bsClass="col"
                      componentClass={[Function]}
                      sm={2}
                      style={
                        Object {
                          "textAlign": "left",
                        }
                      }
                    >
                      mmHG
                    </Col>
                  </FormGroup>
                </Row>
              </Grid>
            }
            formId="bp-form"
            formSubmittedActionCreators={
              Array [
                [Function],
              ]
            }
            title="Blood Pressure"
          >
            <EncounterFormPage
              dispatch={[Function]}
              encounterRole={
                Object {
                  "uuid": "a0b03050-c99b-11e0-9572-0800200c9a66",
                }
              }
              encounterType={
                Object {
                  "uuid": "0C36F6FB-660E-485F-AF04-249579C9EAC9",
                }
              }
              formContent={
                <Grid
                  bsClass="container"
                  componentClass="div"
                  fluid={false}
                >
                  <Section
                    title="Enter Patient Blood Pressure"
                  />
                  <Row
                    bsClass="row"
                    componentClass="div"
                  >
                    <Col
                      bsClass="col"
                      componentClass="div"
                      md={20}
                      sm={20}
                      style={
                        Object {
                          "height": "40px",
                        }
                      }
                    >
                      <span>
                        <h1 />
                      </span>
                    </Col>
                  </Row>
                  <Row
                    bsClass="row"
                    componentClass="div"
                  >
                    <FormGroup
                      bsClass="form-group"
                      controlId="formSystolic"
                    >
                      <Col
                        bsClass="col"
                        componentClass={[Function]}
                        sm={4}
                      >
                        Systolic Blood Pressure
                      </Col>
                      <Col
                        bsClass="col"
                        componentClass="div"
                        sm={2}
                      >
                        <WithFormContext(Connect(Obs))
                          concept="6569bffe-977f-11e1-8993-905e29aff6c1"
                          path="systolic"
                          placeholder="Systolic value"
                          validate={
                            Array [
                              [Function],
                              [Function],
                            ]
                          }
                          warn={[Function]}
                        />
                      </Col>
                      <Col
                        bsClass="col"
                        componentClass={[Function]}
                        sm={2}
                        style={
                          Object {
                            "textAlign": "left",
                          }
                        }
                      >
                        mmHG
                      </Col>
                    </FormGroup>
                  </Row>
                  <Row
                    bsClass="row"
                    componentClass="div"
                  >
                    <FormGroup
                      bsClass="form-group"
                      controlId="formDiastolic"
                    >
                      <Col
                        bsClass="col"
                        componentClass={[Function]}
                        sm={4}
                      >
                        Diastolic Blood Pressure
                      </Col>
                      <Col
                        bsClass="col"
                        componentClass="div"
                        sm={2}
                      >
                        <WithFormContext(Connect(Obs))
                          concept="6569c116-977f-11e1-8993-905e29aff6c1"
                          path="diastolic"
                          placeholder="Diastolic value"
                          validate={
                            Array [
                              [Function],
                              [Function],
                            ]
                          }
                          warn={[Function]}
                        />
                      </Col>
                      <Col
                        bsClass="col"
                        componentClass={[Function]}
                        sm={2}
                        style={
                          Object {
                            "textAlign": "left",
                          }
                        }
                      >
                        mmHG
                      </Col>
                    </FormGroup>
                  </Row>
                </Grid>
              }
              formId="bp-form"
              formSubmittedActionCreators={
                Array [
                  [Function],
                ]
              }
              patient={
                Object {
                  "uuid": "abc",
                  "visit": Object {
                    "uuid": "def",
                  },
                }
              }
              title="Blood Pressure"
            >
              <div
                style={
                  Object {
                    "paddingLeft": "0px",
                    "paddingRight": "0px",
                  }
                }
              >
                <Grid
                  bsClass="container"
                  componentClass="div"
                  fluid={false}
                  style={
                    Object {
                      "paddingLeft": "0px",
                      "paddingRight": "0px",
                    }
                  }
                >
                  <div
                    className="container"
                    style={
                      Object {
                        "paddingLeft": "0px",
                        "paddingRight": "0px",
                      }
                    }
                  >
                    <Row
                      bsClass="row"
                      componentClass="div"
                      style={
                        Object {
                          "backgroundColor": "#ffa500b3",
                        }
                      }
                    >
                      <div
                        className="row"
                        style={
                          Object {
                            "backgroundColor": "#ffa500b3",
                          }
                        }
                      >
                        <Col
                          bsClass="col"
                          componentClass="div"
                          md={20}
                          sm={20}
                          style={
                            Object {
                              "paddingLeft": "20px",
                            }
                          }
                        >
                          <div
                            className="col-md-20 col-sm-20"
                            style={
                              Object {
                                "paddingLeft": "20px",
                              }
                            }
                          >
                            <span>
                              <h1>
                                Blood Pressure
                              </h1>
                            </span>
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <Row
                      bsClass="row"
                      componentClass="div"
                    >
                      <div
                        className="row"
                      >
                        <Col
                          bsClass="col"
                          componentClass="div"
                          md={20}
                          sm={20}
                          style={
                            Object {
                              "height": "40px",
                            }
                          }
                        >
                          <div
                            className="col-md-20 col-sm-20"
                            style={
                              Object {
                                "height": "40px",
                              }
                            }
                          >
                            <span>
                              <h1 />
                            </span>
                          </div>
                        </Col>
                      </div>
                    </Row>
                  </div>
                </Grid>
                <Loader>
                  <div
                    className="custom-loader"
                  >
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                    <div />
                  </div>
                </Loader>
              </div>
            </EncounterFormPage>
          </Connect(EncounterFormPage)>
        </ScreeningForm>
      </Connect(ScreeningForm)>
    </BloodPressureForm>
  </MockRouter>
</Provider>
`;
